ScrapPackedLibrary/ScrapPackedFile.cs

```diff
@@ -243,53 +242,78 @@ private void ExtractFile(string p_packedPath, string p_destinationPath, FileStre
             Directory.CreateDirectory(Path.GetDirectoryName(p_destinationPath));
 
             var fsPacked = p_PackedFileStream;
-            if(fsPacked==null)
+            if (fsPacked == null)
                 fsPacked = new FileStream(fileName, FileMode.Open);
+
+            var fsExtractFile = new FileStream(p_destinationPath, FileMode.Create);
             try
             {
-                var fsExtractFile = new FileStream(p_destinationPath, FileMode.Create);
-                try
-                {
-                    byte[] readBytes = new byte[fileMetaData.FileSize];
+                byte[] readBytes = new byte[fileMetaData.FileSize];
 
-                    fsPacked.Seek(fileMetaData.OriginalOffset, SeekOrigin.Begin);
-                    fsPacked.Read(readBytes, 0, (int)fileMetaData.FileSize);
+                fsPacked.Seek(fileMetaData.OriginalOffset, SeekOrigin.Begin);
+                fsPacked.Read(readBytes, 0, (int)fileMetaData.FileSize);
                     
-                    fsExtractFile.Write(readBytes);
-                }
-                finally
-                {
-                    fsExtractFile.Close();
-                }
+                fsExtractFile.Write(readBytes);
             }
             finally
             {
-                if (p_PackedFileStream == null)
-                    fsPacked.Close();
+                fsExtractFile.Close();
             }
-
-
+            if (p_PackedFileStream == null)
+                fsPacked.Close();
```

Comment:
This needs to be in a finally block …
If an exception happens between opening `fsPacked` and here it needs to get closed anyways …
Now it could be skipped because of an exception.

It may be ok to put it inside the same finally block as `fsExtractFile.Close` …
although what happens if `fsExtractFile.Close` raises an exception? then this might get skipped as well …

I'm not sure why you changed this block of code at all …you did not make any intentional behavioral code change … only removed the nested try blocks which had a reason to be there.